### PR TITLE
feat(stepfunctions-tasks): add EKS call to SFN-tasks

### PR DESCRIPTION
**Pull request by Sumeet-Badyal**
feat(stepfunctions-tasks): support for EKS Call

Implementation

Update package @aws-cdk/aws-stepfunctions-tasks to include support for EKS Call

API as per documentation here:
https://docs.aws.amazon.com/step-functions/latest/dg/connect-eks.html

Includes support for the following Amazon EKS API calls:
eks:call
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  readonly endpoint: string;
+
+  /** The HTTP method ("GET", "POST", "PUT", ...) that clients use to call this method */
+  readonly httpMethod: MethodType;
+
+  /** Path of cluster */
+  readonly path: string;
+
+  /**
+   * Path of cluster
```
Comment (russmackenzie) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts:25]: "Query Parameters part of HTTP request"

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  /** Base 64 encoded certificate data required to communicate with your cluster */
+  readonly certificateAuthority: string;
+
+  /** API endpoint to communicate with your cluster */
+  readonly endpoint: string;
+
+  /** The HTTP method ("GET", "POST", "PUT", ...) that clients use to call this method */
+  readonly httpMethod: MethodType;
+
+  /** Path of cluster */
```
Comment (russmackenzie) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts:21]: "Path part of HTTP request"

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  readonly path: string;
+
+  /**
+   * Path of cluster
+   * @default - no query parameters
+   */
+  readonly queryParameters?: { [key: string]: string[] };
+
+  /**
+   * request body of the http method
```
Comment (russmackenzie) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts:31]: "Request body part of HTTP request"

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  /** Name of the cluster */
+  readonly clusterName: string;
+
+  /** Base 64 encoded certificate data required to communicate with your cluster */
+  readonly certificateAuthority: string;
+
+  /** API endpoint to communicate with your cluster */
+  readonly endpoint: string;
+
+  /** The HTTP method ("GET", "POST", "PUT", ...) that clients use to call this method */
```
Comment (russmackenzie) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts:18]: "HTTP method ("GET", "POST", "PUT", ...) part of HTTP request"

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  /** Delete the resource at the specified endpoint */
+  DELETE = 'DELETE',
+
+  /** Apply partial modifications to the resource */
+  PATCH = 'PATCH',
+
+  /** Retreive data from a server at the specified resource without the response body */
+  HEAD = 'HEAD',
+
+  /** Return data describing what other methods and operations the server supports */
```
Comment (russmackenzie) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts:102]: The EKS integration doesn't include OPTIONS.

**Review by shivlaks — CHANGES_REQUESTED** [review]
similar feedback to the other open PRs, let's align on one first before tackling the rest

**Comment by daisuke-yoshimoto** [comment]
@Sumeet-Badyal @rushimusmaximus @shivlaks 
Will this PR not be merged for a while yet? I would like to use this function.

**Comment by NovakGu** [comment]
Port change to new pull request https://github.com/aws/aws-cdk/pull/12779

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts (hunk 1 above)
Comment (NovakGu) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts:25]: Addressed in new pull request https://github.com/aws/aws-cdk/pull/12779

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts (hunk 2 above)
Comment (NovakGu) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts:21]: Addressed in new pull request https://github.com/aws/aws-cdk/pull/12779

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts (hunk 3 above)
Comment (NovakGu) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts:31]: Addressed in new pull request https://github.com/aws/aws-cdk/pull/12779

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts (hunk 4 above)
Comment (NovakGu) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts:18]: Addressed in new pull request https://github.com/aws/aws-cdk/pull/12779

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts (hunk 5 above)
Comment (NovakGu) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts:102]: Addressed in new pull request https://github.com/aws/aws-cdk/pull/12779

**Comment by NovakGu** [comment]
@daisuke-yoshimoto I'll be taking over the ownership of existing PRs.

**Comment by daisuke-yoshimoto** [comment]
@NovakGu 
Thank you very much. It is very helpful to us.

**Comment by shivlaks** [comment]
superseded by #12779

**Pull request closed without merging**